### PR TITLE
Updating Azure.Communication.NetworkTraversal to new API version 2021-06-21-preview

### PR DESCRIPTION
**Pull request by ajpeacock0**
New API version: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/communication/data-plane/Turn/preview/2021-06-21-preview/CommunicationTurn.json

Additional Changes
- Adds samples to the `Azure.Communication.NetworkTraversal` SDK

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/communication/Azure.Communication.NetworkTraversal/README.md [hunk 1]
```diff
@@ -65,21 +65,21 @@ We guarantee that all client instance methods are thread-safe and independent of
 
 ## Examples
 
-## Generating TURN credentials for a user
+## Getting Relay Configuration for a user
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.NetworkTraversal/README.md:68]:
```suggestion
## Getting a Relay Configuration for a user
```

File: sdk/communication/Azure.Communication.NetworkTraversal/src/CommunicationRelayClientOptions.cs [hunk 2]
```diff
@@ -40,6 +41,10 @@ public enum ServiceVersion
             /// The V2021_02_22_preview1 of the networking service.
             /// </summary>
             V2021_02_22_preview1 = 1,
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.NetworkTraversal/src/CommunicationRelayClientOptions.cs:43]: Should we remove this version? If a developer instantiates the client with this version in the options, what will happen?
Given that the new version is a breaking change and the schema/(de)serializer code has changed, I would expect it to fail.

File: sdk/communication/Azure.Communication.NetworkTraversal/samples/Sample1_CommunicationRelayClientAsync.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+```C# Snippet:CreateCommunicationRelayFromToken
+var endpoint = new Uri("https://my-resource.communication.azure.com");
+TokenCredential tokenCredential = new DefaultAzureCredential();
+var client = new CommunicationRelayClient(endpoint, tokenCredential);
+```
+
+## Get Relay Configuration
+
+The example code snippet below shows how to get a relay configuration for an Azure Communication user that was created following the steps above.
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.NetworkTraversal/samples/Sample1_CommunicationRelayClientAsync.md:37]: > for an Azure Communication user that was created following the steps above.

I can't find the steps that explain how to create a user here.

File: sdk/communication/Azure.Communication.NetworkTraversal/src/autorest.md [hunk 4]
```diff
@@ -7,29 +7,9 @@ Run `dotnet msbuild /t:GenerateCode` to generate code.
 
 ``` yaml
 title: Network traversal
-tag: package-2021-02-22-preview1
+tag: package-2021-06-21-preview
 model-namespace: false
 require:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/60be518b4fa1a9fb011a0cb69ae7ca3e1cee06b1/specification/communication/data-plane/Microsoft.CommunicationServicesTurn/readme.md
+    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/a59747e42da9d74654106f16c5410620b5941fe0/specification/communication/data-plane/Turn/readme.md
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.NetworkTraversal/src/autorest.md:13]: Should the folder in the spec repo be renamed to NetworkTraversal?

File: sdk/communication/Azure.Communication.NetworkTraversal/samples/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+To get started you will need to have an Azure Subscription. Once you have this you can go into the Azure portal and create Azure Communication Services resource. The page will give you necessary information to be able to use the sample codes here such as connections string, shared access key, etc.
+
+This client library allows for the following operations:
+ - Fetch information that allow the bearers to access a TURN server for data relay.
+
+ You can find samples for each of these functions below.
+ - Fetch relay information [synchronously][sample_relay] or [asynchronously][sample_relay_async]
+
+<!-- LINKS -->
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.NetworkTraversal/samples/README.md:24]: I think the links fail the verification check until the PR has been merged, consider commenting them and re-adding them in a follow up PR.

Or maybe changing the url from master to main can help.

https://dev.azure.com/azure-sdk/public/_build/results?buildId=992488&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=72c41882-9d21-5580-4869-7f3ea1bd9c0a

File: sdk/communication/Azure.Communication.NetworkTraversal/samples/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+[sample_relay]: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/communication/Azure.Communication.NetworkTraversal/samples/Sample1_CommunicationRelayClient.md
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.NetworkTraversal/samples/README.md:25]: I assume this needs to be `main` now

```suggestion
[sample_relay]: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/communication/Azure.Communication.NetworkTraversal/samples/Sample1_CommunicationRelayClient.md
```

File: sdk/communication/Azure.Communication.NetworkTraversal/samples/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+[sample_relay_async]: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/communication/Azure.Communication.NetworkTraversal/samples/Sample1_CommunicationRelayClientAsync.md
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.NetworkTraversal/samples/README.md:26]:
```suggestion
[sample_relay_async]: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/communication/Azure.Communication.NetworkTraversal/samples/Sample1_CommunicationRelayClientAsync.md
```

File: sdk/communication/Azure.Communication.NetworkTraversal/CHANGELOG.md [hunk 8]
```diff
@@ -1,7 +1,10 @@
 # Release History
 
-## 1.0.0-beta.2 (Unreleased)
+## 1.0.0-beta.2 (2021-07-09)
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.NetworkTraversal/CHANGELOG.md:3]: This is in the past, are you releasing this week?

**Review by DominikMe — COMMENTED** [review]
Looking good overall, some comments

File: sdk/communication/Azure.Communication.NetworkTraversal/src/CommunicationRelayClientOptions.cs (hunk 2 above)
Comment (ajpeacock0) [reply sdk/communication/Azure.Communication.NetworkTraversal/src/CommunicationRelayClientOptions.cs:43]: Good point! I guess we'd keep the previous SDK version release but the new version wouldn't include the  `2021_06_21_preview` API version

File: sdk/communication/Azure.Communication.NetworkTraversal/src/autorest.md (hunk 4 above)
Comment (ajpeacock0) [reply sdk/communication/Azure.Communication.NetworkTraversal/src/autorest.md:13]: Yea I think that will reduce confusion going forward. I'll put out a PR soon renaming it and see what breaking changes I hit

File: sdk/communication/Azure.Communication.NetworkTraversal/samples/README.md (hunk 5 above)
Comment (ajpeacock0) [reply sdk/communication/Azure.Communication.NetworkTraversal/samples/README.md:24]: I'll try renaming master->main and see if that fixes it

File: sdk/communication/Azure.Communication.NetworkTraversal/samples/Sample1_CommunicationRelayClient.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+```C# Snippet:CreateCommunicationRelayFromToken
+var endpoint = new Uri("https://my-resource.communication.azure.com");
+TokenCredential tokenCredential = new DefaultAzureCredential();
+var client = new CommunicationRelayClient(endpoint, tokenCredential);
+```
+
+## Get Relay Configuration
+
+The example code snippet below shows how to get a relay configuration for an Azure Communication user. For examples on how to create a Azure Communication user, view the [Identity Sampes][identity_samples].
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.NetworkTraversal/samples/Sample1_CommunicationRelayClient.md:37]:
```suggestion
The example code snippet below shows how to get a relay configuration for an Azure Communication user. For examples on how to create a Azure Communication user, view the [Identity Samples][identity_samples].
```

File: sdk/communication/Azure.Communication.NetworkTraversal/samples/Sample1_CommunicationRelayClientAsync.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+```C# Snippet:CreateCommunicationRelayFromToken
+var endpoint = new Uri("https://my-resource.communication.azure.com");
+TokenCredential tokenCredential = new DefaultAzureCredential();
+var client = new CommunicationRelayClient(endpoint, tokenCredential);
+```
+
+## Get Relay Configuration
+
+The example code snippet below shows how to get a relay configuration for an Azure Communication user. For examples on how to create a Azure Communication user, view the [Identity Sampes][identity_samples].
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.NetworkTraversal/samples/Sample1_CommunicationRelayClientAsync.md:37]:
```suggestion
The example code snippet below shows how to get a relay configuration for an Azure Communication user. For examples on how to create a Azure Communication user, view the [Identity Samples][identity_samples].
```

**Review by DominikMe — APPROVED** [review]
Looks good! Found one typo, rest lgtm ✅

**Pull request merged**